sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -35,3 +36,47 @@ case class Size(child: Expression) extends UnaryExpression with ExpectsInputType
     nullSafeCodeGen(ctx, ev, c => s"${ev.primitive} = ($c).size();")
   }
 }
+
+case class ArrayContains(left: Expression, right: Expression) extends BinaryExpression with ExpectsInputTypes {
+  override def dataType: DataType = BooleanType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, AnyDataType)
```

Comment:
My more specific question would be, how should `checkInputDataTypes` look? It seems that the array type and value type should match in "class" of types (eg ints/longs, floats/doubles, perhaps numerics in general, and separately strings etc). I tried using `acceptsType`, but that doesn't seem to be general enough. Should I enumerate all those cases or is there a shortcut?